### ISSUE TITLE
value_type expr checking

### ISSUE THREAD

**Issue by kapilt** [cloud-custodian/cloud-custodian#1617]
at the moment running this policy results in an error, instead of finding all buckets with Name == tag value of App.  What happens when running through is that the value of bucket Name ('xyz') is passed in as the jmespath expression instead of the filter value. There's some unit tests that seem dependent on the extant logic though.

```
policies:
  - name: s3-named
    resource: s3
    filters:
     - type: value
       key: Name
       value: tag:App
       value_type: expr```

**Comment by Sutto** [comment]
This happens because the `value_type` filter is applied to the resource value (specified from the key field), not from the value itself (as far as I can tell) - I need to confirm but I'll have a patch for this shortly.

**Comment by ajkerrigan** [comment]
Looks like this was resolved in #2148, though confirming that revealed a separate edge case (which should be fixed via #6682)
